script.js

```diff
@@ -1,70 +1,96 @@
 (function(global) {
 
-  var requestData = $.getJSON('data.json').done(function(products) {
-    var requestProductTemplate = $.get('product-template.html').done(function(template) {
+  var _template = function(str, param) {
+    return str.replace(/\{(.*?)\}/g, function(match, token) {
+      return param[token];
+    });
+  }
 
-      var data = products;
-      var template = template;
+  var requestData = $.getJSON('data.json').done(function(productData) {
+    var requestProductTemplate = $.get('product-template.html').done(function(productTemplateHtml) {
 
       function TOMApp() {
-        this.products = [];
-      }
-
-      TOMApp.prototype.getproducts = function() {
         var self = this;
-        for (i=0; i<data.sales.length ; i++) {
-          self.products.push(new productobj(data.sales[i], i)  );
+
+        self.regions = {
```

Comment:
Storing DOM selectors of containers (a.k.a Regions) instead of repeating them everywhere. In case we have to modify the html, we can just modify the `regions`. Inspired by Backbone.marionette.